### DISCUSSION TITLE
General Converter - weights and measures conversion plugin

### DISCUSSION THREAD

**Discussion by deefrawley** [Flow-Launcher/Flow.Launcher#602] [Show and tell]
Hi All. 
With the release of Flow 1.8 I've released my second Python plugin - General Converter. It's a simple plugin to convert among various weights and measures. The full list of units is:
<table>
  <tr>
    <th>Subject</th>
    <th>Unit</th>
    <th>Abbreviation</th>
  </tr>
  <tr>
    <td rowspan="8" align="center">Distance</td>
    <td>millimetre</td>
    <td>mm</td>
   </tr>
  <tr>
    <td>centimetre</td>
    <td>cm</td>
  </tr>
  <tr>
    <td>metre</td>
    <td>m</td>
  </tr>
  <tr>
    <td>kilometre</td>
    <td>km</td>
  </tr>
  <tr>
    <td>inch</td>
    <td>in</td>
  </tr>
  <tr>
    <td>foot/feet</td>
    <td>ft</td>
  </tr>
  <tr>
    <td>yard</td>
    <td>yd</td>
  </tr>
  <tr>
    <td>mile</td>
    <td>mi</td>
  </tr>
</table>

<table>
  <tr>
    <th>Subject</th>
    <th>Unit</th>
    <th>Abbreviation</th>
  </tr>
  <tr>
    <td rowspan="9" align="center">Fluids</td>
    <td>millilitre</td>
    <td>ml</td>
   </tr>
  <tr>
    <td>litre</td>
    <td>l</td>
  </tr>
  <tr>
    <td>pint</td>
    <td>pt</td>
  </tr>
  <tr>
    <td>quart</td>
    <td>qt</td>
  </tr>
  <tr>
    <td>cup</td>
    <td>cup</td>
  </tr>
  <tr>
    <td>teaspoon</td>
    <td>tsp</td>
  </tr>
  <tr>
    <td>tablespoon</td>
    <td>tbsp</td>
  </tr>
  <tr>
    <td>gallon</td>
    <td>gal</td>
  </tr>
  <tr>
    <td>gram</td>
    <td>gm</td>
  </tr>
</table>

<table>
  <tr>
    <th>Subject</th>
    <th>Unit</th>
    <th>Abbreviation</th>
  </tr>
  <tr>
    <td rowspan="9" align="center">Area</td>
    <td>square metre</td>
    <td>sqm</td>
   </tr>
  <tr>
    <td>square centremetre</td>
    <td>sqcm</td>
  </tr>
  <tr>
    <td>square kilometre</td>
    <td>sqkm</td>
  </tr>
  <tr>
    <td>square inch</td>
    <td>sqin</td>
  </tr>
  <tr>
    <td>square foot</td>
    <td>sqft</td>
  </tr>
  <tr>
    <td>square mile</td>
    <td>sqmi</td>
  </tr>
  <tr>
    <td>square yard</td>
    <td>sqyd</td>
  </tr>
  <tr>
    <td>hectare</td>
    <td>h</td>
  </tr>
  <tr>
    <td>acre</td>
    <td>ac</td>
  </tr>
</table>

<table>
  <tr>
    <th>Subject</th>
    <th>Unit</th>
    <th>Abbreviation</th>
  </tr>
  <tr>
    <td rowspan="6" align="center">Weight</td>
    <td>gram</td>
    <td>gm</td>
   </tr>
  <tr>
    <td>kilogram</td>
    <td>kg</td>
  </tr>
  <tr>
    <td>pound</td>
    <td>lb</td>
  </tr>
  <tr>
    <td>ounce</td>
    <td>oz</td>
  </tr>
  <tr>
    <td>stone</td>
    <td>st</td>
  </tr>
  <tr>
    <td>ton</td>
    <td>ton</td>
  </tr>
</table>

<table>
  <tr>
    <th>Subject</th>
    <th>Unit</th>
    <th>Abbreviation</th>
  </tr>
  <tr>
    <td rowspan = "2" align="center">Temperature</td>
    <td>celsius</td>
    <td>c</td>
   </tr>
  <tr>
    <td>Farenheit</td>
    <td>f</td>
  </tr>
</table>

More detail [here]( https://github.com/deefrawley/Flow.Launcher.Plugin.GenConvert) . 
If I learn how to set and save settings for python plugins I'll look at a setting for the decimal place accuracy and perhaps the possibility of copying results to the clipboard. Any other feedback, let me know. Thanks

**Comment by Garulf** [comment]
Very cool, if you're still looking to add clipboard capabilities I found [pyperclip](https://pypi.org/project/pyperclip/) to be fairly light-weight.

It basically uses c types for windows clipboard management is better then importing the entire pywin package just for clipboard usage.

**Reply by deefrawley** [reply]
Thanks. I have come across pyperclip but never used it. Will give it a go.

**Comment by deefrawley** [comment]
Just updated to v1.1 with a few usability improvements, bit of a visual upgrade and a few new units.

**Comment by jjw24** [comment]
just used this to convert something on the fly while shopping, very simple to use 👍 💯

**Reply by deefrawley** [reply]
Great to hear others find it useful. Hope you weren't converting kilograms to pounds though as there was a bug (I've just published a fix)!

**Reply by jjw24** [reply]
Nah I wasn't converting that haha
